app/scripts/directives/processTemplateDialog.js

```diff
@@ -4,20 +4,49 @@
   angular.module('openshiftConsole').component('processTemplateDialog', {
     controller: [
       '$scope',
+      '$filter',
+      '$timeout',
+      'Catalog',
       'DataService',
+      'KeywordService',
+      'NotificationsService',
+      'ProjectsService',
+      'RecentlyViewedProjectsService',
       ProcessTemplateDialog
     ],
     controllerAs: '$ctrl',
     bindings: {
       template: '<',
+      project: '<',
+      useProjectTemplate: '<',
       onDialogClosed: '&'
     },
     templateUrl: 'views/directives/process-template-dialog.html'
   });
 
-  function ProcessTemplateDialog($scope, DataService) {
+  function ProcessTemplateDialog($scope,
+                                 $filter,
+                                 $timeout,
+                                 Catalog,
+                                 DataService,
+                                 KeywordService,
+                                 NotificationsService,
+                                 ProjectsService,
+                                 RecentlyViewedProjectsService) {
     var ctrl = this;
     var validityWatcher;
+    var projectListWatcher;
+    var MAX_PROJECTS_TO_WATCH = 250;
+
+    ctrl.selectStep = {
+      id: 'projectTemplates',
+      label: 'Selection',
+      view: 'views/directives/process-template-dialog/process-template-select.html',
+      hidden: ctrl.useProjectTemplate !== true,
```

Comment:
`ctrl.useProjectTemplate !== true` vs `!ctrl.useProjectTemplate`?  Non-boolean options possible?

Reply:
It's a passed in parameter, I am defaulting to false unless explicitly set to true.